src/bp-core/classes/class-bp-user-query.php

```diff
@@ -279,14 +279,16 @@ public function prepare_user_ids_query() {
 				$sql['select']   = "SELECT u.{$this->uid_name} as id FROM {$this->uid_table} u";
 				$sql['where'][]  = $wpdb->prepare( "u.component = %s AND u.type = 'last_activity'", buddypress()->members->id );
 
+				$online_default_time = apply_filters( 'bb_default_online_presence_time', bb_presence_interval() + 5 );
+
 				/**
 				 * Filters the threshold for activity timestamp minutes since to indicate online status.
 				 *
 				 * @since BuddyPress 1.8.0
 				 *
 				 * @param int $value Amount of minutes for threshold. Default 15.
 				 */
-				$sql['where'][] = $wpdb->prepare( 'u.date_recorded >= DATE_SUB( UTC_TIMESTAMP(), INTERVAL %d MINUTE )', apply_filters( 'bp_user_query_online_interval', ( bb_presence_interval() + 5 ) / 60 ) );
+				$sql['where'][] = $wpdb->prepare( 'u.date_recorded >= DATE_SUB( UTC_TIMESTAMP(), INTERVAL %d MINUTE )', $online_default_time / 60 );
```

Comment:
try to use comparison in timestamp directly instead of converting them to minute.